File: packages/android_intent_plus/android/src/main/java/dev/fluttercommunity/plus/androidintent/Bundle/PutClasses/PutBool.java
package dev.fluttercommunity.plus.androidintent.Bundle.PutClasses;

import android.os.Bundle;

import org.json.JSONException;
import org.json.JSONObject;

import dev.fluttercommunity.plus.androidintent.Bundle.Constants;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.base.PutBase;

public class PutBool extends PutBase<Boolean> {

  public final static String JAVA_CLASS = "PutBool";

  public PutBool(String key, boolean value) {
    super(key, JAVA_CLASS, value);
  }

  public static void convert(Bundle bundle, PutBool putBool) {
    bundle.putBoolean(putBool.key, putBool.value);
  }

  public static PutBool fromJson(JSONObject jsonObject) throws JSONException {
    return new PutBool(jsonObject.getString(Constants.KEY), jsonObject.getBoolean(Constants.VALUE));
  }

  @Override
  public JSONObject toJson() throws JSONException {
    JSONObject jsonObject = new JSONObject();
    jsonObject.put("key", key);
    jsonObject.put("javaClass", JAVA_CLASS);
    jsonObject.put("value", value);
    return jsonObject;
  }
}

File: packages/android_intent_plus/lib/put_classes/put_bool.dart
import 'package:android_intent_plus/put_classes/base/put_base.dart';
import 'package:json_annotation/json_annotation.dart';

part 'put_bool.g.dart';

@JsonSerializable()
class PutBool extends PutBase {
  factory PutBool.create({
    required String key,
    required bool value,
  }) =>
      PutBool(
        key: key,
        javaClass: 'PutBool',
        value: value,
      );

  PutBool({
    required String key,
    required String javaClass,
    required this.value,
  }) : super(
          key: key,
          javaClass: javaClass,
        );
  final bool value;

  factory PutBool.fromJson(Map<String, dynamic> json) =>
      _$PutBoolFromJson(json);

  @override
  Map<String, dynamic> toJson() => _$PutBoolToJson(this);
}
